### PR TITLE
Integration tests

### PR DESCRIPTION
**Pull request by danalex97**
I have cherry-picked the integration tests from the unity branch.  The tests start the server and use HTTP requests and keep snapshots of the data received from the server. The snapshots are then compared to see if the rules of the game are respected(for example, players move at most a square each move).

I got rid of all the default values for localhost and made the local client to use the computer's IP. Now multiple users can connect to the same computer even for that client. (these were also necessary for the tests to work properly on Travis) 

The coverage will not increase significantly as coveralls can't actually calculate coverage just from HTTP requests.

### CODE REVIEW

File: aimmo-game/service.py [hunk 1]
```diff
@@ -62,13 +67,34 @@ def player_data(player_id):
         'state': None,
     })
 
+# Plain client routes... These are easy to work with and
+# they are not exposed in the kubernates application
+# as the proxy does not allow communication with them.
+@app.route('/plain/<user_id>/connect')
+def plain_world_init(user_id):
+    world_init()
+    return 'CONNECT'
```
Comment (CelineBoudier) [inline aimmo-game/service.py]: new line please

File: aimmo-game/service.py [hunk 2]
```diff
@@ -62,13 +67,34 @@ def player_data(player_id):
...
+@app.route('/plain/<user_id>/client-ready')
+def plain_client_ready(user_id):
+    world_state = WorldState(state_provider)
+    world_state_manager[int(user_id)] = world_state
+    return 'RECEIVED USER READY ' + user_id
```
Comment (CelineBoudier) [inline aimmo-game/service.py]: new line please

File: aimmo-game/service.py [hunk 3]
```diff
@@ -62,13 +67,34 @@ def player_data(player_id):
...
+@app.route('/plain/<user_id>/exit-game')
+def plain_exit_game(user_id):
+    return "EXITING GAME FOR USER " + user_id
```
Comment (CelineBoudier) [inline aimmo-game/service.py]: new line please

File: aimmo-game/service.py [hunk 4]
```diff
@@ -62,13 +67,34 @@ def player_data(player_id):
         'state': None,
     })
 
+# Plain client routes... These are easy to work with and
+# they are not exposed in the kubernates application
```
Comment (CelineBoudier) [inline aimmo-game/service.py]: kubernetes

File: aimmo-game/tests/test_service.py [hunk 5]
```diff
@@ -1,26 +1,87 @@
...
 
-# TODO: Write test for the new API...
-class TestService(TestCase):
-    def test_healthy(self):
+       avatar = MoveEastDummy(1, Location(0, -1))
+       self.avatars_by_id = {
+           1 : avatar
+       }
+       self.avatars_to_create_by_id = {
+           1 : avatar
```
Comment (CelineBoudier) [inline aimmo-game/tests/test_service.py]: it's the same in to_create and in created, is this relevant?

File: integration-tests/tests/simulation.py [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+        print("Checking player movement...")
+        print(str((last_pos['x'], last_pos['y'])) + " -> " + str((curr_pos['x'], curr_pos['y'])))
+        if x_diff == 0 and y_diff == 0:
+            print("No movement > OK")
+        elif x_move:
+            print("Movement on x axes > OK")
+        elif y_move:
+            print("Movement on y axes > OK")
+        else:
+            print("To many moves > FAIL")
```
Comment (CelineBoudier) [inline integration-tests/tests/simulation.py]: Too many, not to many :) However I wouldn't recommend prints in tests - use asserts please

File: aimmo-game/service.py (hunk 4 above)
Comment (danalex97) [reply aimmo-game/service.py]: =))

File: integration-tests/tests/simulation.py (hunk 6 above)
Comment (danalex97) [reply integration-tests/tests/simulation.py]: :laughing: I'm awful, sorry.

File: aimmo-game/tests/test_service.py (hunk 5 above)
Comment (danalex97) [reply aimmo-game/tests/test_service.py]: I think it does not affect the test, but is confusing, so I will modify it.

**Comment by CelineBoudier** [comment]
Please rebase :)

**Comment by danalex97** [comment]
I rebased.

File: aimmo-game/tests/test_service.py [hunk 7]
```diff
@@ -1,25 +1,83 @@
...
-# TODO: Write test for the new API...
-class TestService(TestCase):
-    def test_healthy(self):
+       avatar = MoveEastDummy(1, Location(0, -1))
+       self.avatars_to_create_by_id = {
+           1 : avatar
+       }
+
+class TestServiceAPI(TestCase):
+    def setUp(self):
```
Comment (CelineBoudier) [inline aimmo-game/tests/test_service.py]: setUp isn't a pep8 name https://www.python.org/dev/peps/pep-0008/#function-names

File: aimmo-game/tests/test_service.py [hunk 8]
```diff
@@ -1,25 +1,83 @@
...
         service.app.config['TESTING'] = True
         self.app = service.app.test_client()
+
+    def test_healthy(self):
         response = self.app.get('/')
         self.assertEqual(response.data, 'HEALTHY')
+
+class TestServiceInternals(TestCase):
+    def setUp(self):
```
Comment (CelineBoudier) [inline aimmo-game/tests/test_service.py]: same

**Pull request merged**